buildkite.sh

```diff
@@ -0,0 +1,356 @@
+      cp --archive --link phobos/etc phobos/std druntime/import/* distribution/imports/
+      cp --archive --link phobos/generated/linux/release/64/libphobos2.{a,so,so*[!o]} distribution/libs/
+      echo '[Environment]' >> distribution/bin/dmd.conf
+      echo 'DFLAGS=-I%@P%/../imports -L-L%@P%/../libs -L--export-dynamic -L--export-dynamic -fPIC' >> distribution/bin/dmd.conf
+
+      tar cvfz distribution.tgz distribution
+    label: "Build"
+    artifact_paths: "distribution.tgz"
+
+  - wait
+EOF
+
+################################################################################
+# Add your project here
+################################################################################
+projects=(
+    "Abscissa/libInputVisitor"
+    "DerelictOrg/DerelictFT"
+    "DerelictOrg/DerelictGL3"
+    "DerelictOrg/DerelictGLFW3"
+    "DerelictOrg/DerelictSDL2"
+    "dlang-bots/dlang-bot"
+    "kaleidicassociates/excel-d"
+    "DlangScience/scid"
+    "Netflix/vectorflow"
+    "PhilippeSigaud/Pegged"
+    "ariovistus/pyd"
+    "atilaneves/automem"
+    "dlang/undeaD"
+    "economicmodeling/containers"
+    "ikod/dlang-requests"
+    "kyllingstad/zmqd"
+    "lgvz/imageformats"
+    "msoucy/dproto"
+    "nomad-software/dunit"
+    "repeatedly/mustache-d"
+    "s-ludwig/std_data_json"
+    "s-ludwig/taggedalgebraic"
+    "libmir/mir"
+    "dlang-community/dfmt"
+    "BBasile/iz"
+    "dlang-community/libdparse"
+    "libmir/mir-random"
+    "dlang-community/D-YAML"
+    "libmir/mir-algorithm"
+    "CyberShadow/ae"
+    "weka-io/mecca"
+    "dlang-community/DCD"
+    "d-gamedev-team/gfm"
+    "atilaneves/unit-threaded"
+    "dlang/tools"
+    "mbierlee/poodinis"
+    "rejectedsoftware/diet-ng"
+    "d-widget-toolkit/dwt"
+    "dlang-tour/core"
+    "dlang-community/D-Scanner"
+    "eBay/tsv-utils-dlang"
+    "BlackEdder/ggplotd"
+    "rejectedsoftware/ddox"
+    "higgsjs/Higgs"
+    "vibe-d/vibe-core+select"
+    "vibe-d/vibe-core+epoll"
+    "dlang/dub"
+    "sociomantic-tsunami/ocean"
+    "vibe-d/vibe.d+libasync-base"
+    "vibe-d/vibe.d+vibe-core-base"
+    "vibe-d/vibe.d+libevent-base"
+    "vibe-d/vibe.d+vibe-core-tests"
+    "vibe-d/vibe.d+libevent-tests"
+    "vibe-d/vibe.d+vibe-core-examples"
+    "vibe-d/vibe.d+libevent-examples"
+)
+
+yaml_echo()
+{
+    echo "      $1"
+}
+
+yaml_indent()
+{
+    # indents for two more characters
+    cat - | sed 's/^/  /'
+}
+
+use_travis_test_script()
+{
+yaml_indent << 'EOF1'
+    cat > travis.py << 'EOF2'
+    import os, sys, yaml
+    def dub_test():
+      print("dub test --compiler=\$DC")
+      sys.exit(0)
+    if not os.path.isfile('.travis.yml'):
+      dub_test()
+    script = yaml.load(open('.travis.yml', 'r')).get('script', '')
+    if isinstance(script, list):
+      script = '\n'.join(script)
+    if len(script) > 0:
+      print(script)
+    else:
+      dub_test()
+    EOF2
+    python travis.py | bash
```

Comment:
You could directly run
```
python << EOF2 | bash

EOF2
```
But here as well, having the script in a file would be simpler.